src/vaadin-custom-field-mixin.html

```diff
@@ -144,9 +141,24 @@
       this.__settingValue = false;
     }
 
-    __setInputsFromChildNodes() {
-      const inputs = Array.from(this.querySelectorAll('*')).filter(node => node.validate || node.checkValidity);
-      this._setInputs(inputs);
+    __getInputsFromSlot() {
+      const isInput = (node => node.validate || node.checkValidity);
+      const slottedElements = Polymer.FlattenedNodesObserver.getFlattenedNodes(this.$.slot)
+        .filter(node => node.nodeType === Node.ELEMENT_NODE);
+      const inputs = [];
+      slottedElements.forEach(elem => {
+        if (isInput(elem)) {
+          inputs.push(elem);
+        } else {
+          const newInputs = Array.from(elem.querySelectorAll('*')).filter(isInput);
```

Comment:
These lines are not currently covered with test. I suppose the `x-field` component in test should handle it?

Reply:
Do you just mean the `else` branch? I'll add tests to cover it.

Reply:
Added tests to cover this too.